### PR TITLE
proof-manager: mock: Check constraint satisfaction in mock

### PR DESCRIPTION
**Pull request by joeykraut**
### Purpose
This PR changes the behavior of the mock proof manager to check constraint satisfaction of the witness and statement. This allows us to check the validity of witness, statement pairs generated by the tasks.

As a result, this PR fixes a few bugs in the task driver integration tests.

### Testing
- Unit tests pass
- Integration tests mostly pass, some error for nitro specific reasons which will be fixed when time permits.

### CODE REVIEW

File: workers/proof-manager/src/mock.rs [hunk 1]
```diff
@@ -31,134 +43,121 @@ pub struct MockProofManager;
...
+        Self::check_constraints::<SizedValidMatchSettle>(&witness, &statement)?;
+
         let proof = dummy_proof();
-        Box::new(GenericMatchSettleBundle { statement, proof })
+        Ok(Box::new(GenericMatchSettleBundle { statement, proof }))
     }
-}
-
-#[cfg(test)]
-mod test {
```
Comment (joeykraut) [inline workers/proof-manager/src/mock.rs:120]: we don't need to test the mock, and it's out of scope now that it requires a valid witness and statement

File: workers/proof-manager/src/mock.rs [hunk 2]
```diff
@@ -31,134 +43,121 @@ pub struct MockProofManager;
...
+        });
     }
 
     /// The execution loop for the mock
-    fn execution_loop(job_queue: &Receiver<ProofManagerJob>) {
+    fn execution_loop(job_queue: &Receiver<ProofManagerJob>) -> Result<(), ProofManagerError> {
         loop {
             match job_queue.recv() {
                 Err(e) => {
                     log::error!("error receiving job: {e}... shutting down");
```
Comment (akirillo) [inline workers/proof-manager/src/mock.rs:58]: should we still log this? seems like the error will necessarily get logged above in `start`. and if we do wanna log here, should we still say "shutting down"?

File: workers/proof-manager/src/mock.rs [hunk 3]
```diff
@@ -31,134 +43,121 @@ pub struct MockProofManager;
...
         };
 
         response_channel.send(bundle).expect(ERR_RESPONSE_CHANNEL_CLOSED);
+        Ok(())
     }
 
     /// Generate a dummy proof of `VALID WALLET CREATE`
     fn valid_wallet_create(
-        _witness: &SizedValidWalletCreateWitness,
+        witness: SizedValidWalletCreateWitness,
```
Comment (akirillo) [inline workers/proof-manager/src/mock.rs:95]: clippy complaining about pass by value here, and for the other mock proof generation methods

**Review by akirillo — APPROVED** [review]
just a couple nits, lgtm

**Pull request merged**